apps/web/src/pages/templates/components/TemplateEditorFormProvider.tsx

```diff
@@ -192,7 +192,10 @@ const TemplateEditorFormProvider = ({ children }) => {
           },
         });
         setTrigger(response.triggers[0]);
-        reset(form);
+        reset({
+          ...form,
+          steps: response.steps,
+        });
```

Comment:
This fixes the issue where multiple duplicate message templates were being created while updating the Notification template from the UI

Steps to reproduce(without this code change):
- Create a new workflow
- add an in-app step, fill the details
-  within the in-app editor screen, click update button
- go back, and add another in-app step
- keep the second in-app editor screen open, click update
- do some minor changes in the content, click update
- Check DB, there will be multiple copies of the 2nd step.

@LetItRock You added this for some digest related issue. can you please confirm if this change works for that issue?